src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h

```diff
@@ -44,11 +114,55 @@ struct typed_primitive_onednn_impl : public typed_primitive_impl<PType> {
           _outer(arg),
           _desc(desc),
           _attrs(attrs),
-          _pd(pd),
-          _prim(pd) { }
+          _pd(pd) {
+            build_primitive();
+        }
 
     bool is_cpu() const override { return false; }
 
+private:
+    std::string get_cache_outpath() const {
+        auto path = _outer.get_program().get_engine().configuration().kernels_cache_path;
+        if (path.empty()) {
+            return {};
+        }
+
+        if (path.back() != '/' && path.back() != '\\') {
+            path += "/";
+        }
+        return path;
+    }
+
+    std::string get_cache_filepath(std::vector<uint8_t> key) const {
+        auto path = get_cache_outpath();
+        if (path.empty()) {
+            return {};
+        }
+
+        std::string key_str(key.begin(), key.end());
+        size_t hash = std::hash<std::string>()(key_str);
+        return path + std::to_string(hash) + ".onednn.cl_cache";
+    }
+
+    void build_primitive() {
+        auto cache_outpath = get_cache_outpath();
+        if (cache_outpath.empty()) {
+            _prim = PrimType(_pd);
+        } else {
+            std::vector<uint8_t> key = _pd.get_cache_blob_id();
+            assert(!key.empty());
+
+            std::vector<uint8_t> cache = load_cache_blob_from_disk(get_cache_filepath(key));
+            if (cache.empty()) {
+                _prim = PrimType(_pd);
+                cache = _prim.get_cache_blob();
+                store_cache_blob_on_disk(get_cache_filepath(key), cache);
+            } else {
+                _prim = PrimType(_pd, cache);
+            }
+        }
+    }
+
```

Comment:
Could you test that it works on windows, too?

Reply:
updated pr